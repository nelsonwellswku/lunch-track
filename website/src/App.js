import React, { Component } from 'react';
import { Grid } from 'react-bootstrap';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import jwtDecode from 'jwt-decode';
import './App.css';
import Home from './components/home/Home';
import MainNav from './components/main-nav/MainNav';
import RegistrationPage from './components/register/RegistrationPage';
import LogInPage from './components/log-in/LogInPage';
import LogOutPage from './components/log-out/LogOutPage';

class App extends Component {
  constructor() {
    super();
    this.state = {
      user: null,
    };

    this.logIn = this.logIn.bind(this);
    this.logOut = this.logOut.bind(this);
  }

  componentWillMount() {
    const authToken = localStorage.getItem('authToken');
    if (authToken) {
      const user = jwtDecode(authToken);
      this.setState({
        user,
      });
    }
  }

  logOut() {
    localStorage.removeItem('authToken');
    this.setState({
      user: null,
    });
  }

  logIn(authToken) {
    localStorage.setItem('authToken', authToken);
    this.setState({
      user: jwtDecode(authToken),
    });
  }

  render() {
    return (
      <div>
        <MainNav user={this.state.user} />

        <Router>
          <Grid>
            <Route exact path="/" component={Home} />
            <Route path="/authentication/login" render={() => <LogInPage logIn={this.logIn} />} />
            <Route path="/authentication/logout" render={() => <LogOutPage logOut={this.logOut} />} />
            <Route path="/authentication/register" component={RegistrationPage} />
          </Grid>
        </Router>
      </div>
    );
  }
}

export default App;
