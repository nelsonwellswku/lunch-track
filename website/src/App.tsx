import React, { useState } from 'react';
import { Route, Switch } from 'react-router-dom';
import MainNav from './components/MainNav';
import AppContext, { IUser } from './contexts/AppContext';
import SignIn from './components/SignIn';
import SignOut from './components/SignOut';
import PageNotFound from './components/PageNotFound';
import Home from './components/Home';
import { AuthClient } from './api/generated';
import appConfig from './appConfig';

const App = () => {
  const existingUserId = window.localStorage.getItem("appUserId");
  const [user, setUser] = useState<IUser | null>(existingUserId ? { appUserId: parseInt(existingUserId, 10) } : null);

  const login = async (token: string) => {

    const client = new AuthClient(appConfig.BaseUrl);
    const response = await client.signIn({
      externalToken: token,
    });

    if (response && response.appUserId) {
      window.localStorage.setItem("appUserId", response.appUserId.toString());
      setUser({ appUserId: response.appUserId });
    }
  };

  const logout = () => {
    const client = new AuthClient(appConfig.BaseUrl);
    client.signOut();
    window.localStorage.removeItem("appUserId");
    setUser(null);
  }

  return (
    <AppContext.Provider value={{
      login,
      logout,
      user,
    }}>

      <MainNav></MainNav>

      <Switch>
        <Route exact path='/'>
          <Home />
        </Route>
        <Route exact path='/sign-in'>
          <SignIn />
        </Route>
        <Route exact path='/sign-out'>
          <SignOut />
        </Route>
        <Route>
          <PageNotFound />
        </Route>
      </Switch>


    </AppContext.Provider>
  );
}

export default App;
