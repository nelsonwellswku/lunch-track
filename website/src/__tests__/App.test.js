import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';

describe('app component', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
