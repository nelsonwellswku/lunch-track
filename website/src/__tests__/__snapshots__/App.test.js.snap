// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "onPostfetch": [Function],
      "onPrefetch": [Function],
      "onUnauthorized": [Function],
      "user": null,
    }
  }
>
  <div>
    <MainNav
      fetching={Object {}}
      user={null}
    />
    <BrowserRouter>
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/authentication/login"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/authentication/logout"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/authentication/register"
          render={[Function]}
        />
        <Route
          path="/authentication/register/verify/:verificationToken"
          render={[Function]}
        />
      </Grid>
    </BrowserRouter>
  </div>
</ContextProvider>
`;
